### PR TITLE
Update to Go 1.12

### LINKED ISSUE #623: Update to Go 1.12
/kind feature
/priority important-soon
/milestone next

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
As per LWKD, Kubernetes master is now on Go 1.12, Cluster API and Cluster API AWS should be updated to match.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
PR #622 addresses this, but have run into issues documented on the PR.

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR updates build tools to Go 1.12

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #623

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/aws/services/elb/loadbalancer.go [hunk 1]
```diff
@@ -312,7 +312,7 @@ func (s *Service) describeClassicELB(name string) (*v1alpha1.ClassicELB, error)
 		}
 	}
 
-	if out == nil && len(out.LoadBalancerDescriptions) == 0 {
+	if out != nil && len(out.LoadBalancerDescriptions) == 0 {
```
Comment (chuckha) [inline pkg/cloud/aws/services/elb/loadbalancer.go]: nice

**Comment by detiber** [comment]
Why the rename here: `pkg/cloud/aws/services/ec2/BUILD → pkg/cloud/aws/services/ec2/BUILD.bazel`

**Comment by vincepri** [comment]
@detiber No reason, I think I deleted and it regenerated the file with the new naming convention Bazel has

**Comment by detiber** [comment]
I thought we standardized on `BUILD` with no suffix because of an issue we were hitting with osx and case sensitivity. @chuckha or @randomvariable do either of you remember?

File: pkg/cloud/aws/services/elb/loadbalancer.go (hunk 1 above)
Comment (vincepri) [reply pkg/cloud/aws/services/elb/loadbalancer.go]: go vet 1.12 is smarter than ever

**Comment by vincepri** [comment]
_Whispers words of encouragement to bazel and mocks_

**Comment by chuckha** [comment]
@detiber It was an edge case we encountered in a vendor directory. The project had a `build/` directory and the `BUILD` file cause chaos on OS X because case insensitivity as you suggested.

Either is fine so long as there is not a conflicting directory.

**Comment by vincepri** [comment]
I thought it was the vice-versa, we're preferring BUILD.bazel to just BUILD for the above reason, wasn't it?

**Comment by chuckha** [comment]
edit; iirc we actually preferred BUILD.bazel for this reason, but it *usually* doesn't matter

**Comment by justaugustus** [comment]
You'll need to bump your CI images to get this to pass.
See here:
- https://github.com/kubernetes/test-infra/pull/12217
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/182#issuecomment-483719222

**Comment by vincepri** [comment]
https://github.com/kubernetes/test-infra/pull/12221

**Comment by randomvariable** [comment]
It was BUILD.bazel to deal with the case insensitivity issue.

We really should rename all the BUILD files to BUILD.bazel for consistency.

Needn't be this PR though (see #722).

/lgtm

**Pull request merged**